### ISSUE TITLE
query no limit causing cassandra to shut down

### ISSUE THREAD

**Issue by evildecay** [Stratio/cassandra-lucene-index#124]
```
ERROR [SharedPool-Worker-2] 2016-04-29 10:29:49,747 JVMStabilityInspector.java:139 - JVM state determined to be unstable.  Exiting forcefully due to:
java.lang.OutOfMemoryError: Requested array size exceeds VM limit
	at java.util.ArrayList.<init>(ArrayList.java:152) ~[na:1.8.0_77]
	at com.stratio.cassandra.lucene.index.FSIndex.lambda$doSearch$0(FSIndex.java:288) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.index.FSIndex$$Lambda$231/1912462331.apply(Unknown Source) ~[na:na]
	at com.stratio.cassandra.lucene.index.FSIndex.doWithSearcher(FSIndex.java:147) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.index.FSIndex.doSearch(FSIndex.java:278) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.index.DocumentIterator.fetch(DocumentIterator.java:68) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.index.DocumentIterator.hasNext(DocumentIterator.java:87) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.IndexReaderWide.prepareNext(IndexReaderWide.java:67) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at com.stratio.cassandra.lucene.IndexReader.hasNext(IndexReader.java:68) ~[cassandra-lucene-index-plugin-3.0.5.0.jar:na]
	at org.apache.cassandra.db.transform.BasePartitions.hasNext(BasePartitions.java:72) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.partitions.UnfilteredPartitionIterators$Serializer.serialize(UnfilteredPartitionIterators.java:295) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.ReadResponse$LocalDataResponse.build(ReadResponse.java:134) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.ReadResponse$LocalDataResponse.<init>(ReadResponse.java:127) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.ReadResponse$LocalDataResponse.<init>(ReadResponse.java:123) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.ReadResponse.createDataResponse(ReadResponse.java:65) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.db.ReadCommand.createResponse(ReadCommand.java:289) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.service.StorageProxy$LocalReadRunnable.runMayThrow(StorageProxy.java:1792) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.service.StorageProxy$DroppableRunnable.run(StorageProxy.java:2445) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[na:1.8.0_77]
	at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$FutureTask.run(AbstractLocalAwareExecutorService.java:164) ~[apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$LocalSessionFutureTask.run(AbstractLocalAwareExecutorService.java:136) [apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at org.apache.cassandra.concurrent.SEPWorker.run(SEPWorker.java:105) [apache-cassandra-3.0.5-SNAPSHOT.jar:3.0.5-SNAPSHOT]
	at java.lang.Thread.run(Thread.java:745) [na:1.8.0_77]
```

**Comment by ealonsodb** [comment]
Hi @evildecay , please take a look at #123

**Comment by adelapena** [comment]
This is fixed in [3.0.5.2](https://github.com/Stratio/cassandra-lucene-index/tree/3.0.5.2), [3.5.2](https://github.com/Stratio/cassandra-lucene-index/tree/3.5.2) and [3.0.6.0](https://github.com/Stratio/cassandra-lucene-index/tree/3.0.6.0) releases.

Thanks for reporting.
